Library/Homebrew/dev-cmd/contributions.rb

```diff
@@ -0,0 +1,101 @@
+# typed: true
+# frozen_string_literal: true
+
+require "cli/parser"
+
+module Homebrew
+  extend T::Sig
+
+  module_function
+
+  SUPPORTED_REPOS = %w[brew core cask bundle].freeze
+
+  sig { returns(CLI::Parser) }
+  def contributions_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `contributions`
+
+        Contributions to Homebrew repos for a user.
+      EOS
+
+      flag "--email=",
+           description: "A user's email address that they commit with."
+
+      flag "--from=",
+           description: "Date (ISO-8601 format) to start searching contributions."
+
+      flag "--to=",
+           description: "Date (ISO-8601 format) to stop searching contributions."
+
+      comma_array "--repos=",
+                  description: "The Homebrew repositories to search for contributions in. " \
+                               "Comma separated. Supported repos: #{SUPPORTED_REPOS.join(", ")}."
+
+      named_args :none
+    end
+  end
+
+  sig { returns(NilClass) }
+  def contributions
+    args = contributions_args.parse
+
+    return ofail "`--repos` and `--email` are required." if !args[:repos] || !args[:email]
```

Comment:
I wonder if `email` (or username/whatever it ends up being) should be a named arg and have `--repos` default to everything. So, I can easily run `brew contributions rslpolster@gmail.com` or `brew contributions Rylan12` to get my complete contribution history.

Reply:
Nice idea, yeah, I like that.